cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -278,12 +276,28 @@ class reader::impl {
   std::optional<std::vector<reader_column_schema>> _reader_column_schema;
   data_type _timestamp_type{type_id::EMPTY};
 
-  // Variables used for chunked reading:
+  // chunked reading happens in 2 parts:
+  //
+  // At the top level there is the "pass" in which we try and limit the
+  // total amount of temporary memory (compressed data, decompressed data) in use
+  // via _input_pass_read_limit.
```

Comment:
Not exactly. Chunks are limited only by the output chunk read limit (the limit on how many bytes of data can be in a chunk).  The only thing that this would affect is the number of chunks you might get back.  For example, if you had a case where the input limit allowed everything to happen in 1 pass, you might get 1 chunk.  But if the input limit caused it to go through 3 passes, you'd get back at least 3 chunks.

Reply:
> if you had a case where the input limit allowed everything to happen in 1 pass, you might get 1 chunk. But if the input limit caused it to go through 3 passes, you'd get back at least 3 chunks.

Chunks can't span passes. This answers my question, thanks.